src/app/services/wallet.service.ts

```diff
@@ -41,60 +42,76 @@ export class WalletService {
     });
   }
 
-  decrypt(passphrase: string, walletEnckey: string): Observable<boolean> {
-    let result = new BehaviorSubject<boolean>(null);
-    let selectedWalletId: string;
-    this.getSelectedWallet().subscribe(
-      (selectedWallet) => (selectedWalletId = selectedWallet.id)
-    );
+  async decrypt(passphrase: string, walletEnckey: string) {
+    var wallet_found: Wallet = await new Promise((resolve) => {
+      this.getSelectedWallet().subscribe((wallet) => {
+        resolve(wallet);
+      });
+    });
 
-    this.syncWallet(selectedWalletId, passphrase, walletEnckey).subscribe(
-      (data) => {
-        if (_.isUndefined(data["result"])) {
-          result.next(false);
-        } else {
-          this.checkWalletBalance(
-            selectedWalletId,
-            passphrase,
-            walletEnckey
-          ).subscribe((data) => {
-            if (_.isNil(data["result"])) {
-              result.next(false);
-            } else {
-              const balance = new BigNumber(data["result"]["total"])
-                .dividedBy("100000000")
-                .toString(10);
-              this.setWalletBalance(balance);
-              this.setDecryptedFlag(true);
-              result.next(true);
-              this.checkWalletAddress(
-                selectedWalletId,
-                passphrase,
-                walletEnckey
-              ).subscribe((data) => {
-                this.setWalletAddress(data["result"][0]);
-              });
-              this.checkWalletViewKey(
-                selectedWalletId,
-                passphrase,
-                walletEnckey
-              ).subscribe((data) => {
-                this.setWalletViewKey(data["result"]);
-              });
-              this.checkWalletTxnHistory(
-                selectedWalletId,
-                passphrase,
-                walletEnckey
-              ).subscribe((data) => {
-                this.setWalletTxnHistory(data["result"]);
-              });
-            }
-          });
-        }
-      }
-    );
+    let selectedWalletId = wallet_found.id;
+
+    var data = await this.syncWallet(
+      selectedWalletId,
+      passphrase,
+      walletEnckey
+    ).toPromise();
+
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+
+    if (_.isUndefined(data["result"])) {
+      return false;
+    }
+
+    await this.refresh(passphrase, walletEnckey);
+    return true;
+  }
+
+  async refresh(passphrase, walletEnckey) {
+    var wallet_found: Wallet = await new Promise((resolve) => {
+      this.getSelectedWallet().subscribe((wallet) => {
+        resolve(wallet);
+      });
+    });
```

Comment:
Similarly here

Reply:
ok

Reply:
direct await for BehaviourSubject doesn't work correctly as expected.
so added converting function